### DISCUSSION TITLE
SASView

### DISCUSSION THREAD

**Discussion by Nolla2024** [SasView/sasview#3094] [Q&A] [answered]
Can someone please guide me through step-by-step process of preforming a linear fit and guinier analysis using Sasview and extracting Rg. I am new to this.

**Comment by smk78** [comment, marked as answer]
@Nolla2024 , sorry we haven't responded sooner. We are unacustomed to receiving requests for assistance through Github discussions! Emailing help@sasview.org is the usual route.

So, first of all, if you have not done so already, please take the time to read our ['Getting Started'](https://www.sasview.org/downloads/getting_started_with_sasview_v5.pdf) and ['Basic 1D Fitting'](https://www.sasview.org/downloads/basic_1d_fitting_in_sasview_v5.pdf) tutorials.

To be honest, performing a traditional Guinier analysis of the form you enquire about is probably best considered a deprecated methodolgy these days, as it is so subjective and more robust alternatives exist.

But to try and answer your question...

You cannot simply load data and fit in the manner you want because you need to fit in Ln(I) vs Q^2 space and not I vs Q space which is what SasView is intended to do. So we have to convert the input data into something SasView can deal with. You _could_, of course, do that outside of the program and load the converted data and then use SasView as normal. But here is another way.

1. load your data (here I'm picking the conalbumin test data shipped with SasView) and then click (bottom left) Plot > Create New
![image](https://github.com/user-attachments/assets/49da1fef-40ca-40b7-b3d2-4f4d71549a8b)

2. Right-click on Graph1 & select Change Scale
![image](https://github.com/user-attachments/assets/4dadef9a-21d8-446c-8af6-0aea881c8b7f)

3. Select Guinier from the View drop-down & OK
![image](https://github.com/user-attachments/assets/8882762b-78c4-49b4-a8e2-6082b42967f3)

4. Now we must expand the Guinier region (the subjective part). There are a couple of ways to do this, for example, right-clicking on the Graph and Toggle Navigation menu will bring up a magnifying glass tool, or you can right-click and Set Graph Range. Here I've chosen 0.0 < Q^2 < 0.005.
![image](https://github.com/user-attachments/assets/29ee0fc1-4605-4f01-a648-6b04693834cd)

5. Right-click on the Graph again, but this time select the name of the dataset; a sub-menu will appear. Select Linear Fit
![image](https://github.com/user-attachments/assets/95391437-1ca2-4ecf-a55c-b18c0995b9a4)

6. Enter the range of Q^2 you want to fit over and click Fit
![image](https://github.com/user-attachments/assets/cb75bfae-f759-40a8-9ebc-0bafa5da855b)

HOWEVER...

There is another way to analyse the input data directly,

7. Send the loaded data for fitting
![image](https://github.com/user-attachments/assets/4657445c-5af5-473f-9af0-970c8f4df2c0)

8. And then in the FitPage select the guinier model from the Shape-Independent category
![image](https://github.com/user-attachments/assets/fc3acdab-956f-4ca5-9322-46d8e6153932)

9. Click Compute/Plot
![image](https://github.com/user-attachments/assets/58e6849f-bebc-44de-a529-c5ac88ff7ec3)

10. In the Fit Options tab ente rthe range of Q values to fit. In the linear fit above I used 0.0005 < Q^2 < 0.005, so that would be 0.022 < Q < 0.07
![image](https://github.com/user-attachments/assets/b0513b3b-342a-4dff-8f5a-90789e69ed45)

11. Return to the Model tab and check all 3 parameters.
![image](https://github.com/user-attachments/assets/501bbde6-4ef3-46ed-b19a-74bd0e07cf3b)

12. Click Fit
![image](https://github.com/user-attachments/assets/59e2d9b3-c8a1-4df6-85e2-a8854eb7d78b)

So the difference between these approaches is that the first method is fitting a m.Q^2+C straight line to linearised data, whereas the second is fitting the actual [Guinier function](https://www.sasview.org/docs/user/models/guinier.html) to the experimental data. One method gives Rg~29 Ang, the other ~33 Ang.

You pay your money, you take your choice.

Another way to estimate Rg is to perform P(r) inversion of the data. [Also possible in SasView](https://www.sasview.org/downloads/pr_inversion_analysis_in_sasview_v5.pdf).
